### PR TITLE
nodetaint: Bump k8s dependencies

### PR DESCRIPTION
**Pull request by pdeslaur**
Bump a number of Kubernetes packages to reduce the CVE count.

### CODE REVIEW

**Comment by pdeslaur** [comment]
I'm not sure anymore about the `-dropreplace` here... I may just force a specific version instead

File: nodetaint.yaml [hunk 1]
```diff
@@ -22,18 +22,44 @@ pipeline:
       expected-commit: 9f648717bd31a782696335450ee05a5ed79cd870
 
   - runs: |
-      # Mitigate  GHSA-3vm4-22fp-5rfm, GHSA-gwc9-m7rh-j2ww and GHSA-8c26-wmh5-6g9v
+      # Mitigate GHSA-3vm4-22fp-5rfm, GHSA-gwc9-m7rh-j2ww and GHSA-8c26-wmh5-6g9v
```
Comment (vaikas) [inline nodetaint.yaml:25]: Can't comment on L29, but that's woefully out of date.
https://pkg.go.dev/golang.org/x/text?tab=versions

Comment (pdeslaur) [reply nodetaint.yaml:25]: Yeah... The upstream repo has an old version, so believe it or not, but v0.3.8 is still an update: https://github.com/wish/nodetaint/blob/master/go.sum#L655

I prefer to not always "upgrade to latest" unless we need to - but happy to bump it!

**Review by vaikas — APPROVED** [review]
Thank you!!!

File: nodetaint.yaml (hunk 1 above)
Comment (vaikas) [reply nodetaint.yaml:25]: Yeah, makes sense.

**Pull request merged**